server/src/parse-excel-file.ts

```
@@ -0,0 +1,18 @@
+import XLSX from 'xlsx';
+
+export async function parseExcelFile(filePath: string) {
+    const workbook = XLSX.readFile(filePath);
+    const sheetName: string = workbook.SheetNames[0] || "";
+    const sheet = workbook.Sheets[sheetName];
+    if (!sheet) return [];
+  
+    const jsonData: any[] = XLSX.utils.sheet_to_json(sheet, { header: 1 });
+  
+    const dataWithoutFirstRow = jsonData.slice(1);
+  
+    if (dataWithoutFirstRow.length > 0) {
+      dataWithoutFirstRow[0] = dataWithoutFirstRow[0].slice(3);
+    }
```

if this is empty you should throw an error that says no timescale data to parse

---

also can i ask why you `slice` here?

---

Because I am getting rid of the first 3 columns since they are just black in the excel sheet